performant-software/DM
# Firefox readonly stabilization

### What does this PR do?
Introduces some fixes initiated in #177, but rolls these back to focus on supporting readonly behavior for non-Chrome browsers while minimizing alteration to (stable in Chrome) text editor functionality. Adds an alert modal when a user is logged in through a non-Chrome browser.

### What issues does it address?
Closes #173, #174, #175, #176 (in part by removing support for editor functionality in non-Chrome browsers)

### How to test
- Using Chrome, visit http://sandbox.digitalmappa.org and create + view a project with a few text and canvas resources to regression test, with a particular focus on text editor behavior. See list of side-effect bugs in #177 and check whether any of the bugs that are not Firefox-specific can be reproduced.
- Test readonly (not logged in) viewing for public projects at http://sandbox.digitalmappa.org . Then log in and check for the appearance of a warning popup re: lack of support for editing in non-Chrome browsers.
